packages/utils/events/src/Disposable.js

```diff
@@ -33,21 +33,20 @@ export default class Disposable implements IDisposable {
     }
   }
 
-  dispose(): void {
+  async dispose(): Promise<void> {
     if (this.disposed) {
       return;
     }
 
+    this.disposed = true;
+
     invariant(this.#disposables != null);
-    for (let disposable of this.#disposables) {
-      if (typeof disposable === 'function') {
-        disposable();
-      } else {
-        disposable.dispose();
-      }
-    }
+    await Promise.all(
```

Comment:
Should `DisposableLike` and `IDisposable` rather return `Async<mixed>`? Would at least make the intention obvious in the API